src/internal/channel.js

```diff
@@ -113,7 +113,7 @@ export function channel(buffer) {
 }
 
 export function eventChannel(subscribe, buffer = buffers.none(), matcher) {
-  if(arguments.length > 1)
+  if(typeof matcher !== 'undefined')
```

Comment:
It's a valid POV but I personally think using `undefined` is not a good idea for indicating no args as it can overlaps with the so common undefined Errors (maybe also my typed lang. Java/FP background). i agree with your point on the `buffer` arg;  I considered making it mandatory or requiring a `null` instead of `undefined` but didn't like it much from the API viewpoint. So I just settled with the no-buffer solution. Trade-offs as usual :).